### ISSUE TITLE
PCP_TMPFILE_DIR used at build time as well as run time

### ISSUE THREAD

**Issue by ddiss** [performancecopilot/pcp#1140]
This appears to be a regression from 029add427a42344ea7129452c9e0fb286d3cf881 .

When configured with default `prefix`, `localstatedir` and `tmpdir` paths, the build phase attempts to create temporary files under a non existent `/usr/local/var/tmp/` path, resulting in a failed build.
I think usage of PCP_TMPFILE_DIR should be limited to run-time only, or alternatively `prefix` should be removed from the `--with-tmpdir` default.

**Comment by kmcdonell** [comment]
Trying a build --with-tmpdir=/deadbeef and repeating until it works.
So far I've found the following places:
src/libpcp/src/mk.exports
src/libpcp/src/mk.pmdbg
src/pmieconf/xtractnames
src/pmns/Make.stdpmid

And in each, replaced the tmp=... line with a stanza like this:
if [ -n "$PCP_TMPFILE_DIR" -a -d "$PCP_TMPFILE_DIR" ]
then
    tmp=`mktemp -d "$PCP_TMPFILE_DIR/pcp-foo.XXXXXXXXX"` || exit 1
else
    # assume run during a build and /tmp is a safe bet
    #
    tmp=`mktemp -d "/tmp/pcp-foo.XXXXXXXXX"` || exit 1
fi

I plan to commit this and await feedback from @ddiss

**Comment by ddiss** [comment]
@kmcdonell thanks for looking at this!
https://github.com/kmcdonell/pcp/commit/adb769a8a9a56659774bc127a0bb7bc3527e4d25 looks good to me, although it could be changed to also consider `$TMPDIR` for the build-time case. E.g.
```
if [ -n "$PCP_TMPFILE_DIR" -a -d "$PCP_TMPFILE_DIR" ]
then
        tmp=`mktemp --tmpdir="$PCP_TMPFILE_DIR" -d pmns_stdpmid.XXXXXXXXX` || exit 1
else
        # assume run during a build, use $TMPDIR or /tmp.
        #
        tmp=`mktemp --tmpdir -d pmms_stdpmid.XXXXXXXXX` || exit 1
fi
```

**Comment by kmcdonell** [comment]
Thanks @ddiss, but unfortunately mktemp --tmpdir does not work at all in non-GNU land, like the *BSD bretheren,

**Comment by ddiss** [comment]
> Thanks @ddiss, but unfortunately mktemp --tmpdir does not work at all in non-GNU land, like the *BSD bretheren

in that case, please feel free to submit as-is. I'll give it a spin on our build service.

**Comment by ddiss** [comment]
> in that case, please feel free to submit as-is. I'll give it a spin on our build service.

@kmcdonell: JFYI, I'm carrying your patch downstream as a fix for this ticket. Please let me know when you think it'll be merged.

**Comment by kmcdonell** [comment]
Thanks @ddiss.  I overlooked closing this issue.
This was merged as commit adb769a8a on 29 Jan, and would have made into the recent PCP 5.2.4 release.
